packages/react-app/src/components/GiveawayAsideContent.tsx

```diff
@@ -0,0 +1,31 @@
+import { useContext } from 'react';
+
+import { Typography } from '@mui/material';
+
+import { Giveaway } from '../lib/types';
+import { GiveawayContext } from '../pages/details';
+
+const GiveawayAsideContent = () => {
+  const giveaway = useContext(GiveawayContext) as Giveaway;
+
+  return giveaway.rules ? (
```

Comment:
Since this component will not render anything if the rules are not present isn't it better to have that validation outside and not render the component at all if the rules are not present?